### ISSUE TITLE
"use client" needs to be placed before other expressions

### ISSUE THREAD

**Issue by djshubs** [ndom91/react-timezone-select#117]
### Describe the bug

Not sure what is happening, but the compilation is cause the "use client" to be placed below.

```shell
Failed to compile.

./node_modules/react-timezone-select/dist/index.js
Error: 
  x The "use client" directive must be placed before other expressions. Move it to the top of the file to resolve this issue.
     ,-[/{REDACTED}/node_modules/react-timezone-select/dist/index.js:119:1]
 119 | 
 120 | // src/index.tsx
 121 | import { jsx } from "react/jsx-runtime";
 122 | "use client";
     : ^^^^^^^^^^^^^
 123 | function useTimezoneSelect({
 124 |   timezones = timezone_list_default,
 125 |   labelStyle = "original",
     `----

Import trace for requested module:
./node_modules/react-timezone-select/dist/index.js
```

### Reproduction

I got the error by updating from 3.0.2 to 3.2.2

### System Info

```Shell
System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (16) x64 Common KVM processor
    Memory: 42.36 GB / 47.04 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 19.2.0 - ~/.nvm/versions/node/v19.2.0/bin/node
    Yarn: 3.4.1 - ~/.nvm/versions/node/v19.2.0/bin/yarn
    npm: 9.6.6 - ~/.nvm/versions/node/v19.2.0/bin/npm
    pnpm: 8.14.1 - ~/.nvm/versions/node/v19.2.0/bin/pnpm
```

### Used Package Manager

npm

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ndom91** [comment]
Ah okay I know what's wrong, thanks for pointing this out, will get to this shortly.

**Comment by ndom91** [comment]
Hey @djshubs this should be fixed in `3.2.3`, thanks!

**Comment by djshubs** [comment]
> Hey @djshubs this should be fixed in `3.2.3`, thanks!

Just built with the new release and it built successfully. Thanks for a quick fix @ndom91!
